### PR TITLE
parallel_sort for graph loading functionality

### PR DESCRIPTION
**Pull request by orazve**
parallel_sort functionality is used during load graph. It improves graph converting performance.

### CODE REVIEW

File: cpp/daal/src/algorithms/service_qsort.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/* file: service_qsort.h */
+/*******************************************************************************
+* Copyright 2014-2020 Intel Corporation
```
Comment (averbukh) [inline cpp/daal/src/algorithms/service_qsort.h:3]: 2020

**Review by averbukh — COMMENTED** [review]
Minor comment only

File: cpp/daal/src/algorithms/service_qsort.h [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /* file: service_qsort.h */
 /*******************************************************************************
-* Copyright 2014-2020 Intel Corporation
+* Copyright 2020-2020 Intel Corporation
```
Comment (averbukh) [inline cpp/daal/src/algorithms/service_qsort.h:3]: Just "Copyright 2020 Intel Corporation". See the rest of the library and follow for consistency.

File: cpp/daal/src/externals/core_threading_win_dll.cpp [hunk 3]
```diff
@@ -277,8 +277,8 @@ typedef void (*_daal_tbb_task_scheduler_free_t)(void *& globalControl);
 typedef size_t (*_setNumberOfThreads_t)(const size_t, void **);
 typedef void * (*_daal_threader_env_t)();
 
-typedef void (*_daal_parallel_sort_int_t)(int *, int *);
-typedef void (*_daal_parallel_sort_size_t_t)(size_t *, size_t *);
+typedef void (*_daal_parallel_sort_int32_t)(std::int32_t *, std::int32_t *);
+typedef void (*_daal_parallel_sort_uint64_t)(std::uint64_t *, std::uint64_t *);
```
Comment (averbukh) [inline cpp/daal/src/externals/core_threading_win_dll.cpp:281]: In fact we prefer to use int and size_t in DAAL.

File: cpp/daal/src/threading/threading.h [hunk 4]
```diff
@@ -25,6 +25,7 @@
 #define __THREADING_H__
 
 #include "services/daal_defines.h"
+#include <cstdint>
```
Comment (averbukh) [inline cpp/daal/src/threading/threading.h:28]: It isn't look like we need this dependency here. In fact you only need to have 'unsigned long *' and 'unsigned long long *' casting in the safe way.

**Review by averbukh — CHANGES_REQUESTED** [review]
In your particular case, the introduction of additional dependency to standard library is not required.

File: cpp/daal/src/externals/core_threading_win_dll.cpp (hunk 3 above)
Comment (orazve) [reply cpp/daal/src/externals/core_threading_win_dll.cpp]: fixed

File: cpp/daal/src/threading/threading.h (hunk 4 above)
Comment (orazve) [reply cpp/daal/src/threading/threading.h]: removed

**Review by averbukh — APPROVED** [review]

**Pull request merged**